diff --git a/res/values/strings.xml b/res/values/strings.xml
index 7fd027a863..6e3c7a8c28 100644
--- a/res/values/strings.xml
+++ b/res/values/strings.xml
@@ -32,7 +32,9 @@

     <!-- Device Info --> <skip />
     <!-- Device Info screen. Used for a status item's value when the proper value is not known -->
-    <string name="device_info_default">Unknown</string>
+    <string name="device_info_default">MT6795</string>
+    <!-- Device Info screen. Used for a status item's value when the proper value is not known -->
+    <string name="author_info_default">DarkAbhi</string>
     <!-- [CHAR LIMIT=NONE] Device Info screen. Countdown for user taps to enable development settings -->
     <plurals name="show_dev_countdown">
         <item quantity="one">You are now <xliff:g id="step_count">%1$d</xliff:g> step away from being a developer.</item>
@@ -2375,6 +2377,8 @@
     <string name="firmware_version">Android version</string>
     <!-- About phone screen, status item label  [CHAR LIMIT=60] -->
     <string name="security_patch">Android security patch level</string>
+    <!-- About phone screen, status item label  [CHAR LIMIT=60] -->
+    <string name="author_info">Build author</string>
     <!-- About phone screen, status item label  [CHAR LIMIT=40] -->
     <string name="model_info">Model</string>
     <!-- About phone screen, fcc equipment id label  [CHAR LIMIT=40] -->
diff --git a/res/xml/device_info_settings.xml b/res/xml/device_info_settings.xml
index c032d882e8..d360c00499 100644
--- a/res/xml/device_info_settings.xml
+++ b/res/xml/device_info_settings.xml
@@ -137,6 +137,14 @@
                 <intent android:action="android.intent.action.VIEW"
                         android:data="https://source.android.com/security/bulletin/" />
         </PreferenceScreen>
+
+        <!-- Author Info -->
+        <PreferenceScreen android:key="author_info"
+                android:title="@string/author_info"
+                android:summary="@string/author_info_default">
+                <intent android:action="android.intent.action.VIEW"
+                        android:data="http://abhishekan.me" />
+        </PreferenceScreen>

         <!-- Device FCC equipment id -->
         <com.android.settings.DividerPreference
diff --git a/src/com/android/settings/DeviceInfoSettings.java b/src/com/android/settings/DeviceInfoSettings.java
index 4ae69941e5..6684de17b7 100644
--- a/src/com/android/settings/DeviceInfoSettings.java
+++ b/src/com/android/settings/DeviceInfoSettings.java
@@ -77,6 +77,7 @@ public class DeviceInfoSettings extends SettingsPreferenceFragment implements In
     private static final String KEY_BASEBAND_VERSION = "baseband_version";
     private static final String KEY_FIRMWARE_VERSION = "firmware_version";
     private static final String KEY_SECURITY_PATCH = "security_patch";
+    private static final String KEY_AUTHOR_INFO = "author_info";
     private static final String KEY_UPDATE_SETTING = "additional_system_update_settings";
     private static final String KEY_EQUIPMENT_ID = "fcc_equipment_id";
     private static final String PROPERTY_EQUIPMENT_ID = "ro.ril.fccid";
@@ -135,7 +136,6 @@ public class DeviceInfoSettings extends SettingsPreferenceFragment implements In

         setValueSummary(KEY_BASEBAND_VERSION, "gsm.version.baseband");
         setValueSummary(KEY_EQUIPMENT_ID, PROPERTY_EQUIPMENT_ID);
-        setStringSummary(KEY_DEVICE_CPU, DeviceInfoUtils.getDeviceProcessorInfo(getActivity()));
         setStringSummary(KEY_DEVICE_MODEL, Build.MODEL);
         setStringSummary(KEY_BUILD_NUMBER, Build.DISPLAY);
         setValueSummary(KEY_MOD_BUILD_DATE, "ro.build.date");
@@ -163,6 +163,10 @@ public class DeviceInfoSettings extends SettingsPreferenceFragment implements In
         removePreferenceIfPropertyMissing(getPreferenceScreen(), KEY_MBN_VERSION,
 		PROPERTY_MBN_VERSION);

+
+           String author_info = getResources().getString(R.string.author_info_default);
+           setStringSummary(KEY_AUTHOR_INFO, author_info);
+
         if (!SELinux.isSELinuxEnabled()) {
             String status = getResources().getString(R.string.selinux_status_disabled);
             setStringSummary(KEY_SELINUX_STATUS, status);
@@ -340,6 +344,13 @@ public class DeviceInfoSettings extends SettingsPreferenceFragment implements In
             }
         } else if (preference.getKey().equals(KEY_BUILD_TYPE)) {
             sendFeedback();
+        } else if (preference.getKey().equals(KEY_AUTHOR_INFO)) {
+            if (getPackageManager().queryIntentActivities(preference.getIntent(), 0).isEmpty()) {
+                // Don't send out the intent to stop crash
+                Log.w(LOG_TAG, "Stop click action on " + KEY_AUTHOR_INFO + ": "
+                        + "queryIntentActivities() returns empty" );
+                return true;
+            }
         } else if (preference.getKey().equals(KEY_DEVICE_FEEDBACK)) {
             sendFeedback();
         } else if(preference.getKey().equals(KEY_SYSTEM_UPDATE_SETTINGS)) {
